src/database/DatabaseBackend.h

```diff
@@ -2,8 +2,30 @@
 #include "DBOperation.h"
 #include "config/ConfigVariable.h"
 
-extern ConfigGroup db_backend_config;
-extern ConfigVariable<std::string> db_backend_type;
+
+#if NEW_BACKEND_CONFIG
```

Comment:
If this is going out as a "major" version bump, then it would make sense to ignore backwards compatibility. However, if this is only a "minor" bump then it's safe (and probably a better practice) to leave support for older configuration files.